dart-impl/base/src/logging.c

```diff
@@ -22,3 +22,29 @@ const int dart__base__unit_term_colors[DART_LOG_TCOL_NUM_CODES-1] = {
   DART_LOG_TCOL_RED,
   DART_LOG_TCOL_BLUE
 };
+
+
+enum dart__base__logging_loglevel
+dart__base__logging_env_loglevel()
+{
+  static enum dart__base__logging_loglevel level = DART_LOGLEVEL_TRACE;
+  static int log_level_parsed = 0;
+  if (!log_level_parsed) {
+    const char *envstr = getenv(DART_LOGLEVEL_ENVSTR);
+    if (envstr) {
```

Comment:
I agree that `dart_config` does not present a well-defined interface, but parsing environment settings in several unrelated places is a bad thing to do.
The implementation of `dash::util::Config` is way more comprehensive than the current snippet in DART.
DART should be considered a stand-alone deployable, I agree, so it should provide its own run-time config interface. Parsing `ENV` is not required, however.
If at all, `getopt` should be used for this.

So I propose to extend dart-if by a configuration interface and use it in `dash::util::Config` to apply environment settings.